tendrl/monitoring_integration/graphite/__init__.py

```diff
@@ -0,0 +1,153 @@
+import os
+import copy
+
+
+import etcd
+import time
+import socket
+
+
+from tendrl.commons.utils import etcd_utils
+from tendrl.commons.utils import log_utils as logger
+from tendrl.monitoring_integration.grafana import cluster_detail
+
+
+class GraphitePlugin():
+
+    def __init__(self):
+
+        self.host = NS.config.data["datasource_host"]
+        self.port = NS.config.data["datasource_port"]
+        self.carbon_port = NS.config.data["carbon_port"]
+        self.prefix = 'tendrl'
+        self._connect()
+
+    def _connect(self):
+
+        try:
+            self.graphite_sock = socket.socket()
+            self.graphite_sock.connect((self.host, int(self.carbon_port)))
+        except socket.error as ex:
+            logger.log("error", NS.get("publisher_id", None),
+                       {'message': "Cannot connect to graphite socket" + str(ex)})
+            raise ex
+            
+
+    def push_metrics(self, metric_name, metric_value):
+
+        message = '%s%s%s %s %d\n' % (
+            self.prefix,
+            str("."),
+            metric_name,
+            str(metric_value),
+            int(time.time())
+        )
+        
+        response = self.graphite_sock.sendall(message)
```

Comment:
Catch errors in sendall and , if connection issue, retry sending message again just once